src/models/bgp/community.rs

```diff
@@ -112,6 +113,38 @@ pub enum ExtendedCommunity {
     Raw([u8; 8]),
 }
 
+impl ExtendedCommunity {
+    pub const fn community_type(&self) -> Option<ExtendedCommunityType> {
+        match self {
+            ExtendedCommunity::TransitiveTwoOctetAsSpecific(_) => {
+                Some(ExtendedCommunityType::TransitiveTwoOctetAsSpecific)
+            }
+            ExtendedCommunity::TransitiveIpv4AddressSpecific(_) => {
+                Some(ExtendedCommunityType::TransitiveIpv4AddressSpecific)
+            }
+            ExtendedCommunity::TransitiveFourOctetAsSpecific(_) => {
+                Some(ExtendedCommunityType::TransitiveFourOctetAsSpecific)
+            }
+            ExtendedCommunity::TransitiveOpaque(_) => Some(ExtendedCommunityType::TransitiveOpaque),
+            ExtendedCommunity::NonTransitiveTwoOctetAsSpecific(_) => {
+                Some(ExtendedCommunityType::NonTransitiveTwoOctetAsSpecific)
+            }
+            ExtendedCommunity::NonTransitiveIpv4AddressSpecific(_) => {
+                Some(ExtendedCommunityType::NonTransitiveIpv4AddressSpecific)
+            }
+            ExtendedCommunity::NonTransitiveFourOctetAsSpecific(_) => {
+                Some(ExtendedCommunityType::NonTransitiveFourOctetAsSpecific)
+            }
+            ExtendedCommunity::NonTransitiveOpaque(_) => {
+                Some(ExtendedCommunityType::NonTransitiveOpaque)
+            }
+            // TODO: Does Ipv6AddressSpecific not have a format type?
+            ExtendedCommunity::Ipv6AddressSpecific(_) => None,
```

Comment:
I believe the IPv6 will have the same types as IPv4, but we will need to double check.

https://www.rfc-editor.org/rfc/rfc5701.html#section-2

```

    0                   1                   2                   3
    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   | 0x00 or 0x40  |    Sub-Type   |    Global Administrator       |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |          Global Administrator (cont.)                         |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |          Global Administrator (cont.)                         |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |          Global Administrator (cont.)                         |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   | Global Administrator (cont.)  |    Local Administrator        |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+

   The first high-order octet indicates whether a particular sub-type of
   this community is transitive across Autonomous Systems (ASes) (0x00),
   or not (0x40).  The second high-order octet of this extended type is
   used to indicate sub-types.  The sub-types are the same as for the
   IPv4 Address Specific Extended Community.
   ```

Reply:
I read into it a bit, but it seems like IPv6 extended communities are actually a separate attribute. Perhaps it would be better to make it a separate struct and add it to `MetaCommunity`?

Reply:
I think it makes more sense to make it a `ExtendedCommunity` rather than a separate `MetaCommunity`.  We can probably use the original implementation

```rust
#[derive(Debug, PartialEq, Clone, Copy, Eq)]
#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
pub struct Ipv6AddressSpecific {
    pub ec_type: u8,
    pub ec_subtype: u8,
    // 16 octets
    pub global_administrator: Ipv6Addr,
    // 2 octets
    pub local_administrator: [u8; 2],
}
```

Or extend it a bit to make it `Transitive`/`NonTransitive` to match the IPv4 communities for symmetry.